treewide: fix various dtc warnings

DragonBluep opened the pull request:
These patches have not been tested. Most warnings are caused by `#address-cells` and `#size-cells`.

hnyman commented:
> These patches have not been tested.

It sounds pretty daring to propose untested changes to 80+ DTS files.

Some of the proposed changes look pretty harmless (removing leading zeroes, adding missing address-cells), but some other changes might possibly change actual functionality.

Would it be better to split the PR into two, and do just the easy changes first, and then the others in a second PR. Hopefully after somebody has tested them.

musashino205 commented:
`ipq8064: ` ---> `ipq806x: ` (1e85e0d)

namiltd commented on target/linux/ramips/dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:
```diff
@@ -48,12 +48,12 @@
 		compatible = "jedec,spi-nor";
 		reg = <0>;
-		#address-cells = <1>;
-		#size-cells = <1>;
```
Adding these lines removes the warning:
"OF: Bad cell count for /palmbus@10000000/spi@b00/flash@0/partitions"

If you delete them the warning will come back.

DragonBluep replied:
This warning is an upstream bug. The child node `partition` doesn't have `reg` property, we should not add `#address-cells` and `#size-cells` here.

namiltd replied:
How else can we fix this bug?

DragonBluep replied:
It's harmless. If you are interested, you can try tracking the source code.

KanjiMonster replied:
I'd rather have some pointless lines (dtc doesn't complain about them, right?) than a user-visible error/warning on every boot.

Once this is fixed upstream, we can remove them; but until then I'd rather keep them in.

DragonBluep replied:
> I'd rather have some pointless lines (dtc doesn't complain about them, right?) than a user-visible error/warning on every boot.

Though dtc did't complain about it, it seems that the driver has done some address translation operations. I'm not sure if it's really harmless. ref: https://elixir.bootlin.com/linux/v6.6.82/source/drivers/of/address.c#L592

If it's really annoying, I'd prefer just silence the printf(): https://github.com/openwrt/openwrt/pull/11302/commits/e74219c795a03e8b9adccbb5e6a20d4566872f8a

> 
> Once this is fixed upstream, we can remove them; but until then I'd rather keep them in.

Anyway, I'll revert this change.

DragonBluep commented:
> `ipq8064: ` ---> `ipq806x: ` ([1e85e0d](https://github.com/openwrt/openwrt/commit/1e85e0d5483c6338fb17d864d0c6fa8b5f74a49c))

@musashino205 Fixed. Thanks.

Gingernut1978 commented:
Warning on the Totolink A8000RU:

```
[    0.506082] mtk-ecc 1100e000.ecc: probed
[    0.511375] spi-nand spi0.0: Macronix SPI NAND was found.
[    0.516807] spi-nand spi0.0: 128 MiB, block size: 128 KiB, page size: 2048, OOB size: 64
[    0.524940] mtk-snand 1100d000.spi: ECC strength: 4 bits per 512 bytes
[    0.534492] [BBT] BMT.v2 is found at 0x3ff
[    0.538886] 8 fixed-partitions partitions found on MTD device spi0.0
[    0.545248] Creating 8 MTD partitions on "spi0.0":
[    0.550041] 0x000000000000-0x000000080000 : "Preloader"
[    0.556177] 0x000000080000-0x0000000c0000 : "ATF"
[    0.561436] 0x0000000c0000-0x000000140000 : "u-boot"
[    0.567314] 0x000000140000-0x0000001c0000 : "u-boot-env"
[    0.573448] 0x0000001c0000-0x000000200000 : "factory"
[    0.579090] OF: Bad cell count for /spi@1100d000/flash@0/partitions
[    0.585576] 0x000000200000-0x000006600000 : "ubi"
[    0.695477] 0x000006600000-0x000006700000 : "User_data"
[    0.702046] 0x000006700000-0x0000075c0000 : "reserved"
```

DragonBluep commented:
> Is this warning on the Totolink A8000RU the same as the one on the ruijie_rg-ew1200g-pro-v1.1?

@Gingernut1978 Yes, a lot of devices have this warning. However, this PR doesn't  solve such issue. We only focus on dtc (device-tree-compiler) warning here.

Djfe commented:
Thanks a lot for cleaning up all of these files :)

Offtopic, I had an idea:
Would it be possible to add some check to the Github Action for PRs that reports these dtc warnings for changed dts/dtsi files? Or is that way too verbose?

DragonBluep commented:
> Would it be possible to add some check to the Github Action for PRs that reports these dtc warnings for changed dts/dtsi files? Or is that way too verbose?

Not yet, some warnings cannot be fixed. They are caused by some necessary workaround. For example:
https://github.com/openwrt/openwrt/blob/324f234966fe7b910c7f25b69334e54591098c18/target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-rt-ax89x.dts#L505-L515

DragonBluep committed:
qualcommax: fix switch node dtc warnings for Asus RT-AX89X

Add the missing #address-cells and #size-cells to fix the following
dtc warnings:

ipq8074-rt-ax89x.dts:558.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@0:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:566.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@1:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:572.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@2:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:578.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@3:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:584.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@4:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:590.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@5:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)
ipq8074-rt-ax89x.dts:596.5-15: Warning (reg_format): /soc@0/mdio@90000/switch@10/ports/port@6:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 1)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
ath79: dts: fix wifi node name for Fortinet FAP-221-C

The DTC recommends using wifi@0,0 as the node name to match the reg
property structure. Fix warning:

../dts/qca9557_fortinet_fap-221-c.dts:208.13-213.4: Warning (pci_device_reg): /ahb/pcie@180c0000/wifi@0,0,0: PCI unit address format error, expected "0,0"

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
apm821xx: dts: explicitly set the partition reg size for Netgear WNDR4700

Correct the mtd partition reg property size to address the following
dtc warnings:

../dts/netgear-wndr4700.dts:191.6-33: Warning (reg_format): /plb/opb/ebc/ndfc@1,0/nand/partitions/partition@0/partition@40000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
mvebu: fix partition node dtc warnings for Synology DS213j

Add missing #address-cells and #size-cells to fix the
following dtc warnings:

armada-370-synology-ds213j.dts:288.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@0:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
armada-370-synology-ds213j.dts:294.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@c0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
armada-370-synology-ds213j.dts:299.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@100000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
armada-370-synology-ds213j.dts:304.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@110000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
armada-370-synology-ds213j.dts:308.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@7d0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
armada-370-synology-ds213j.dts:327.5-35: Warning (reg_format): /soc/spi@10600/flash@0/partitions/partition@7e0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
kirkwood: dts: explicitly set the partition reg size for Blackarmor NAS220

Correct the mtd partition reg property size to address the following
dtc warnings:

kirkwood-blackarmor-nas220.dts:185.4-24: Warning (reg_format): /mbus@f1000000/nand@12f/partitions/partition@0:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
kirkwood-blackarmor-nas220.dts:191.4-28: Warning (reg_format): /mbus@f1000000/nand@12f/partitions/partition@a0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
kirkwood-blackarmor-nas220.dts:197.4-28: Warning (reg_format): /mbus@f1000000/nand@12f/partitions/partition@b0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
kirkwood-blackarmor-nas220.dts:203.4-30: Warning (reg_format): /mbus@f1000000/nand@12f/partitions/partition@c0000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
mvebu: fix usb node dtc warnings for Ctera C200 V2

Correct #address-cells and #size-cells based on child node reg
property structure to fix the following dtc warnings:

armada-370-c200-v2.dts:342.6-16: Warning (reg_format): /soc/pcie@82000000/pcie@1,0/bridge@0,1/usb@1,0/port@1:reg: property has invalid length (4 bytes) (#address-cells == 3, #size-cells == 2)
armada-370-c200-v2.dts:347.6-16: Warning (reg_format): /soc/pcie@82000000/pcie@1,0/bridge@0,1/usb@1,0/port@2:reg: property has invalid length (4 bytes) (#address-cells == 3, #size-cells == 2)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
qoriq: dts: add missing reg property for WatchGuard Firebox M300

Set the reg property value based on node name "pca9547@77". This
patch fixes the following dtc warning:

watchguard-firebox-m300.dts:364.14-366.5: Warning (i2c_bus_reg): /soc@ffe000000/i2c@118000/pca9547@77: missing or empty reg property

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
mvebu: dts: fix unit name leading 0s warnings

Trim unnecessary 0s from the node name to fix the dtc warnings:

cn9131-puzzle-m901.dts:43.18-46.4: Warning (unit_address_format): /memory@00000000: unit name should not have leading 0s
cn9130-clearfog-pro.dts:33.18-36.4: Warning (unit_address_format): /memory@00000000: unit name should not have leading 0s
cn9132-puzzle-m902.dts:50.18-53.4: Warning (unit_address_format): /memory@00000000: unit name should not have leading 0s
armada-385-wd_cloud-mirror-gen2.dts:148.26-152.9: Warning (unit_address_format): /soc/internal-regs/nand-controller@d0000/nand@0/partitions/partition@00000000: unit name should not have leading 0s
armada-385-wd_cloud-mirror-gen2.dts:154.26-157.9: Warning (unit_address_format): /soc/internal-regs/nand-controller@d0000/nand@0/partitions/partition@00500000: unit name should not have leading 0s
armada-385-wd_cloud-mirror-gen2.dts:159.26-163.9: Warning (unit_address_format): /soc/internal-regs/nand-controller@d0000/nand@0/partitions/partition@00a00000: unit name should not have leading 0s
armada-385-wd_cloud-mirror-gen2.dts:165.26-168.9: Warning (unit_address_format): /soc/internal-regs/nand-controller@d0000/nand@0/partitions/partition@00f00000: unit name should not have leading 0s

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
ramips: dts: correct WiFi band node reg size for Gemtek WVRTM-130ACN

The #address-cells should be 1 according to the dt-binding document.
This patch fixes the following dtc warnings:

../dts/mt7621_gemtek_wvrtm-130acn.dts:46.4-14: Warning (reg_format): /pcie@1e140000/pcie@0,0/wifi@0,0/band@0:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 0)
../dts/mt7621_gemtek_wvrtm-130acn.dts:54.4-14: Warning (reg_format): /pcie@1e140000/pcie@0,0/wifi@0,0/band@1:reg: property has invalid length (4 bytes) (#address-cells == 2, #size-cells == 0)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
ramips: dts: explicitly set the partition reg size for Ruijie RG-EW1200G

Correct the mtd partition reg property size to address the following
dtc warnings:

../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:60.5-30: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@0:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:66.5-29: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@50000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:72.5-29: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@60000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:88.5-29: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@70000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:94.5-29: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@80000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)
../dts/mt7621_ruijie_rg-ew1200g-pro-v1.1.dts:101.5-30: Warning (reg_format): /palmbus@1e000000/spi@b00/flash@0/partitions/partition@90000:reg: property has invalid length (8 bytes) (#address-cells == 2, #size-cells == 1)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

DragonBluep committed:
bmips: fix dtc warnings for D-Link DSL-2750B

- Add the missing ranges property for PCIe bridge node.
- Correct the PCIe device node name.

This patch fix the following dtc warning:

../dts/bcm6328-dlink-dsl-2750b-b1.dts:203.9-220.4: Warning (pci_bridge): /ubus/pcie@10e40000/pcie@0: missing ranges for PCI bridge (or not a bridge)

Signed-off-by: Shiji Yang <yangshiji66@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/18242
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

neheb commented:
ping @Noltari

Noltari reviewed (approved):
LGTM (bmips changes)
